### PR TITLE
i3wsr: init at 1.3.1

### PR DESCRIPTION
**Pull request by SEbbaDK**
###### Motivation for this change
Packaging a simple way to rename i3 and sway workspaces via `i3wsr`and `swaywsr`.
Packaged together because they depend on the same core and are built in nearly identical ways.
The software builds fine on macOS, but it doesn't make much sense to use it as neither WM works on macOS.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8601,6 +8601,12 @@
     githubId = 19472270;
     name = "Sebastian";
   };
+  sebbadk = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8604]: Please put this into a separate commit with the message ``maintainers: add sebbadk``.

File: pkgs/applications/window-managers/sway/wsr.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  nativeBuildInputs = [ python3 ];
+  buildInputs = [ libxcb ];
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Automatically change sway workspace names based on their contents";
+    longDescription = ''
+      Automatically sets the workspace names to match the windows on the workspace. The chosen name for a workspace is a composite of the app_id or WM_CLASS X11 window property for each window in a workspace.
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/sway/wsr.nix:24]: Please break the line after ~100 characters.

File: pkgs/applications/window-managers/sway/wsr.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    rev = "6c4671c702f647395d983aaf607286db1c692db6";
+    sha256 = "0bmpbhyvgnbi5baj6v0wdxpdh9cnlzvcc44vh3vihmzsp6i5q05a";
+  };
+
+  cargoSha256 = "15wa03279lflr16a6kw7zcn3nvalnjydk9g6nj7xqxmc7zkpf0rm";
+
+  nativeBuildInputs = [ python3 ];
+  buildInputs = [ libxcb ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/sway/wsr.nix:19]: Why did you disable tests?

File: pkgs/applications/window-managers/i3/wsr.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    rev = "v${version}";
+    sha256 = "1zpyncg29y8cv5nw0vgd69nywbj1ppxf6qfm4zc6zz0gk0vxy4pn";
+  };
+
+  cargoSha256 = "0snys419d32anf73jcvrq8h9kp1fq0maqcxz6ww04yg2jv6j47nc";
+
+  nativeBuildInputs = [ libxcb python3 ];
+  buildInputs = [ libxcb ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/i3/wsr.nix:19]: Why did you disable tests?

File: pkgs/applications/window-managers/i3/wsr.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  meta = with lib; {
+    description = "Automatically change i3 workspace names based on their contents";
+    longDescription = ''
+      Automatically sets the workspace names to match the windows on the workspace. The chosen name for a workspace is a composite of the WM_CLASS X11 window property for each window in a workspace.
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/i3/wsr.nix:24]: Please break the line after ~100 characters.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/window-managers/i3/wsr.nix (hunk 4 above)
Comment (SEbbaDK) [reply pkgs/applications/window-managers/i3/wsr.nix:19]: Upstream release has tests failing

File: pkgs/applications/window-managers/sway/wsr.nix (hunk 3 above)
Comment (SEbbaDK) [reply pkgs/applications/window-managers/sway/wsr.nix:19]: Upstream release has tests failing

File: pkgs/applications/window-managers/i3/wsr.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchFromGitHub {
+    owner = "roosta";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1zpyncg29y8cv5nw0vgd69nywbj1ppxf6qfm4zc6zz0gk0vxy4pn";
+  };
+
+  cargoSha256 = "0snys419d32anf73jcvrq8h9kp1fq0maqcxz6ww04yg2jv6j47nc";
+
+  nativeBuildInputs = [ libxcb python3 ];
```
Comment (zseri) [inline pkgs/applications/window-managers/i3/wsr.nix:16]: Are you sure something from libxcb is executed at build-time?
```suggestion
  nativeBuildInputs = [ python3 ];
```

Comment (SEbbaDK) [reply pkgs/applications/window-managers/i3/wsr.nix:16]: Just checked again, and it doesn't

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117377` at c362a3b6 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>i3wsr</li>
    <li>swaywsr</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `stdenv`.
Near pkgs/applications/window-managers/i3/wsr.nix:1:8:
```
  |
1 | { lib, stdenv, fetchFromGitHub, rustPlatform, libxcb, python3 }:
  |        ^
```
</li>
    <li>warning: unused-argument

Unused argument: `stdenv`.
Near pkgs/applications/window-managers/sway/wsr.nix:1:8:
```
  |
1 | { lib, stdenv, fetchFromGitHub, rustPlatform, libxcb, python3 }:
  |        ^
```
</li>
  </ul>
</details>

File: pkgs/applications/window-managers/i3/wsr.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/applications/window-managers/i3/wsr.nix:19]:
```suggestion
  # has no tests
  doCheck = false;
```

File: pkgs/applications/window-managers/sway/wsr.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/applications/window-managers/sway/wsr.nix:19]:
```suggestion
  # has no tests
  doCheck = false;
```

File: pkgs/applications/window-managers/i3/wsr.nix (hunk 4 above)
Comment (SEbbaDK) [reply pkgs/applications/window-managers/i3/wsr.nix:20]: Included in newest commits

File: pkgs/applications/window-managers/sway/wsr.nix (hunk 3 above)
Comment (SEbbaDK) [reply pkgs/applications/window-managers/sway/wsr.nix:20]: Included in newest commits

**Review by fogti — APPROVED** [review]
